test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandler.Extensibility.IssuerSigningKey.cs

```diff
@@ -0,0 +1,284 @@
+                var utcNow = DateTime.UtcNow;
+                var utcPlusOneHour = utcNow + TimeSpan.FromHours(1);
+
+                #region return CustomIssuerSigningKeyValidationError
+                // Test cases where delegate is overridden and return a CustomIssuerSigningKeyValidationError
+                // CustomIssuerSigningKeyValidationError : IssuerSigningKeyValidationError, ExceptionType: SecurityTokenInvalidIssuerSigningKeyException
+                theoryData.Add(new IssuerSigningKeyExtensibilityTheoryData(
+                    "CustomIssuerSigningKeyValidatorDelegate",
+                    utcNow,
+                    CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorDelegate,
+                    extraStackFrames: 2)
+                {
+                    ExpectedException = new ExpectedException(
+                        typeof(SecurityTokenInvalidSigningKeyException),
+                        nameof(CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorDelegate)),
+                    IssuerSigningKeyValidationError = new CustomIssuerSigningKeyValidationError(
+                        new MessageDetail(
+                            nameof(CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorDelegate), null),
+                        ValidationFailureType.SigningKeyValidationFailed,
+                        typeof(SecurityTokenInvalidSigningKeyException),
+                        new StackFrame("CustomIssuerSigningKeyValidationDelegates.cs", 160),
+                        null)
+                });
+
+                // CustomIssuerSigningKeyValidationError : IssuerSigningKeyValidationError, ExceptionType: CustomSecurityTokenInvalidIssuerSigningKeyException : SecurityTokenInvalidIssuerSigningKeyException
+                theoryData.Add(new IssuerSigningKeyExtensibilityTheoryData(
+                    "CustomIssuerSigningKeyValidatorCustomExceptionDelegate",
+                    utcNow,
+                    CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorCustomExceptionDelegate,
+                    extraStackFrames: 2)
+                {
+                    ExpectedException = new ExpectedException(
+                        typeof(CustomSecurityTokenInvalidSigningKeyException),
+                        nameof(CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorCustomExceptionDelegate)),
+                    IssuerSigningKeyValidationError = new CustomIssuerSigningKeyValidationError(
+                        new MessageDetail(
+                            nameof(CustomIssuerSigningKeyValidationDelegates.CustomIssuerSigningKeyValidatorCustomExceptionDelegate), null),
+                        ValidationFailureType.SigningKeyValidationFailed,
+                        typeof(CustomSecurityTokenInvalidSigningKeyException),
+                        new StackFrame("CustomIssuerSigningKeyValidationDelegates.cs", 175),
```

Comment:
Do we assert or use this line number anywhere? My initial thought is that this can cause flaky tests. Plus every time an internal refactor is done, these test cases can be affected. Maybe something to discuss.

Reply:
The number is included because the constructor for `StackFrame` requires file name and line number.
Because of the reason you mentioned regarding flaky tests, we are not asserting this number anywhere.

I am considering simply setting the number to 0 to avoid confusion.